types/enginko/MCF-LW12CO2E/uplink.js

```diff
@@ -838,12 +876,35 @@ function consume(event) {
     if (isNumeric(value)) {
       value = Number(value);
     }
-    sample[variable] = value;
+
+    // If theres a second date, it implicates that a second sample will get emited
+    if (
```

Comment:
the logic still seems a bit confusing to me - I would already decide when to create new samples in the for loop looping over content. basically each time you see a date, you will need a new sample

Reply:
Not all samples have a date so i need to check for that too ^^' . Hope i coded it clearer now

Reply:
what is the format of the message if the second sample does not have a date? how can this happen..? ^^